matron/src/screen_results.c

```diff
@@ -0,0 +1,31 @@
+#include <semaphore.h>
+#include <stdio.h>
+
+#include "events.h"
+#include "screen_results.h"
+
+// instead of a queue, we just have a single event here!
+// i think for the moment it is enough, 
+// since only one blocking request can be in flight at a time
+// note that we only store a pointer, 
+// and consumer must free the event data
+static union event_data* ev_result;
+static sem_t sem_result;
+
+void screen_results_init() {
+    sem_init(&sem_result, 0, 0);
+}    
+
```

Comment:
Might be good practice to have a `screen_results_deinit()` called in main's `cleanup()` function. Can free the event data in the very slim chance cleanup() get's called while blocked on a request.

Reply:
yes, good call